homeassistant/helpers/trigger.py

```diff
@@ -49,6 +60,100 @@
     "pluggable_actions"
 )
 
+TRIGGER_DESCRIPTION_CACHE: HassKey[dict[str, dict[str, Any]]] = HassKey(
+    "trigger_description_cache"
+)
+ALL_TRIGGER_DESCRIPTIONS_CACHE: HassKey[dict[str, dict[str, Any]]] = HassKey(
+    "all_trigger_descriptions_cache"
+)
+TRIGGER_PLATFORM_SUBSCRIPTIONS: HassKey[
+    list[Callable[[set[str]], Coroutine[Any, Any, None]]]
+] = HassKey("trigger_platform_subscriptions")
+TRIGGERS: HassKey[dict[str, str]] = HassKey("triggers")
+
+
+_FIELD_SCHEMA = vol.Schema(
+    {},
+    extra=vol.ALLOW_EXTRA,
+)
+
+_TRIGGER_SCHEMA = vol.Schema(
+    {
+        vol.Optional("fields"): vol.Schema({str: _FIELD_SCHEMA}),
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def starts_with_dot(key: str) -> str:
+    """Check if key starts with dot."""
+    if not key.startswith("."):
+        raise vol.Invalid("Key does not start with .")
+    return key
+
+
+_TRIGGERS_SCHEMA = vol.Schema(
+    {
+        vol.Remove(vol.All(str, starts_with_dot)): object,
```

Comment:
maybe I missed something about the triggers, but why is this needed? what are the dot prefixed keys?